code/modules/emote_panel/emote_panel.dm

```diff
@@ -0,0 +1,62 @@
+/datum/emote_panel
+	var/list/blacklisted_emotes = list("me", "help")
+
+/datum/emote_panel/ui_static_data(mob/user)
+	var/list/data = list()
+
+	var/list/emotes = list()
+	var/list/keys = list()
+
+	for(var/key in GLOB.emote_list)
+		for(var/datum/emote/emote in GLOB.emote_list[key])
+			if(emote.key in keys)
+				continue
+			if(emote.key in blacklisted_emotes)
+				continue
+			if(emote.can_run_emote(user, status_check = FALSE, intentional = FALSE))
+				keys += emote.key
+				emotes += list(list(
+					"key" = emote.key,
+					"name" = emote.name,
+					"emote_path" = emote.type,
+					"hands" = emote.hands_use_check,
+					"visible" = emote.emote_type & EMOTE_VISIBLE,
+					"audible" = emote.emote_type & EMOTE_AUDIBLE,
+					"sound" = !isnull(emote.get_sound(user)),
+					"use_params" = emote.message_param,
+				))
+
+	data["emotes"] = emotes
+
+	return data
+
+/datum/emote_panel/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+	if(.)
+		return
+	switch(action)
+		if("play_emote")
+			var/emote_path = params["emote_path"]
+			var/use_params = params["use_params"]
+			var/datum/emote/emote = new emote_path()
```

Comment:
This code is an enormous vulnerability and allows a crafty user to spawn any object in the entire game. jlsnow caught this but incorrectly doubted themselves. This needs to be fixed. Your PR has been reverted for this reason